docs/concepts/joins.rst

```diff
@@ -350,6 +350,32 @@ optimized.*
    Complex queries are broken down into subqueries that are run on their shards
    before joining.
 
+.. _join-optim-eliminate-cross-joins:
+
+Cross join elimination
+----------------------------
+
+The optimizer will try to eliminate cross joins in the query plan by changing
+the join-order. When reordering, it will try to preserve the original join order
+as much as possible. If a cross join cannot be eliminated, the original join
+order will be maintained. It will try to eliminate any cross join
+it can e.g.:
+
+.. code-block:: SQL
+
+    SELECT *
+    FROM t1 CROSS JOIN t2
+    INNER JOIN t3
+    ON t3.z = t1.x AND t3.z = t2.y
+
+The cross join elimination will change the order of the query from t1, t2, t3
+to t2, t1, t3 so that each join has a join condition and the cross-join can be
+eliminated. This optimizer rule can be disabled with the session settings::
+
+    SET optimizer_eliminate_cross_join = false
+
+Note that this setting is experimental, and may change in the future.
```

Comment:
I think the setting should be documented in docs/config/session.rst and this should link to it / vice versa

Reply:
👍